lib/tasks/after_tasks.rake

```diff
@@ -631,6 +631,28 @@ namespace :After do
     end
     puts && STDOUT.flush
   end
+
+  desc "Add user_id to existing logged in kudos without user_ids"
+  task(add_user_id_to_kudos: :environment) do
+    kudos_to_update = Kudo.where("pseud_id IS NOT NULL AND user_id IS NULL")
+    kudos_count = kudos_to_update.count
+    total_batches = kudos_count / 1000
+    puts "Total number of kudos to check: #{kudos_count}"
+
+    kudos_to_update.find_in_batches.with_index do |batch, index|
```

Comment:
lol, that was my initial thought in the wee hours of Sunday morning:

> sarken  05:45
> I was thinking about https://otwarchive.atlassian.net/browse/AO3-5851, and since we only need/want to update the user_id column for users and pseuds that exist... would it make more sense to loop through all of our _users_, get their kudos (user.kudos does the trick), and then update -- no, never mind, as I'm saying this I'm realizing that would just mean going through the entire kudos table once per user

But kudos are indexed on pseuds, so maybe that would be better?

Reply:
You could do `Kudo.where(pseud_id: user.pseud_ids).update_all(user_id: user.id)` or some such.

(I think the only user where that query might be unreasonable for is `orphan_account`, but I don't think the `orphan_account` can even leave kudos, so you can probably skip it if that's a concern.)

Reply:
I've checked and there are no kudos from `orphan_account`, so that shouldn't be a problem.

Reply:
Hmm. The code for adding the `user_id` when new kudos are created will be deployed before this. Should we be selecting _all_ of the user's kudos, or just the ones that don't have a `user_id`? I feel like skipping the ones with the `user_id` is probably not worth the trouble, considering the new code will only have been live for a couple weeks when this is run.

Reply:
It's not too hard to add `.where(user_id: nil)` to whatever the kudos query is. But I'm guessing it wouldn't affect the speed too much either way, so maybe it's just down to preference? IDK.